charts/argocd-progressive-rollout/templates/psp.yaml

```diff
@@ -0,0 +1,57 @@
+{{- if .Values.rbac.pspEnabled }}
+apiVersion: policy/v1beta1
+kind: PodSecurityPolicy
+metadata:
+  name: {{ template "argocd-progressive-rollout.fullname" . }}
+  labels:
+    {{- include "argocd-progressive-rollout.labels" . | nindent 4 }}
+  annotations:
+    seccomp.security.alpha.kubernetes.io/allowedProfileNames: '*'
+spec:
+  privileged: false
+  hostIPC: false
+  hostNetwork: false
+  hostPID: false
+  readOnlyRootFilesystem: false
+  allowPrivilegeEscalation: false
+  allowedCapabilities:
+    - '*'
+  fsGroup:
+    rule: RunAsAny
+  runAsUser:
+    rule: RunAsAny
+  seLinux:
+    rule: RunAsAny
+  supplementalGroups:
+    rule: RunAsAny
+  volumes:
+    - '*'
+---
+kind: ClusterRole
```

Comment:
Should we be granting a cluster wide role or scope it to a namespace instead?

Reply:
I was porting the Kustomize version to Helm as per https://github.com/Skyscanner/argocd-progressive-rollout/blob/main/config/rbac/role.yaml
When you use the kubebuilder annotation it creates a ClusteRole as per https://book.kubebuilder.io/reference/markers/rbac.html

Do you have any concern?

Reply:
Just trying to see if we can give the least permissions as possible to the operator. We will be monitoring a single argocd namespace, right so technically we don't need a ClusterRole? I am fine with keeping it as a ClusterRole if that is what kubebuilder recommends.

Reply:
Since the RBAC is generated by the kubebuilder itself, I don't see a way to change it looking at https://book.kubebuilder.io/reference/markers/rbac.html